pyresample/future/geometry/swath.py

```diff
@@ -19,4 +19,40 @@
 
 from __future__ import annotations
 
-from pyresample.geometry import CoordinateDefinition, SwathDefinition  # noqa
+from pyresample.geometry import CoordinateDefinition  # noqa
+from pyresample.geometry import SwathDefinition as LegacySwathDefinition
+
+
+class SwathDefinition(LegacySwathDefinition):
+    """Swath defined by lons and lats.
+
+    Parameters
+    ----------
+    lons : numpy array
+    lats : numpy array
+    crs: pyproj.CRS,
+       The CRS to use. longlat on WGS84 by default.
+    nprocs : int, optional
+        Number of processor cores to be used for calculations.
+    attrs: dict,
+       A dictionary made to store metadata.
+
+    Attributes
+    ----------
+    shape : tuple
+        Swath shape
+    size : int
+        Number of elements in swath
+    ndims : int
+        Swath dimensions
+    lons : object
+        Swath lons
+    lats : object
+        Swath lats
+    cartesian_coords : object
+        Swath cartesian coordinates
+    """
+
+    def __init__(self, lons, lats, crs=None, nprocs=1, attrs=None):
```

Comment:
Please remove `nprocs`. The commit message says "for backwards compatibility". There is no backwards compatibility in the future modules...at least let's try really hard to avoid it. If it is going to be here then it at least needs a warning or something about it being deprecated. But I'd rather have a `create_swath_def` pytest fixture that excludes `nprocs` from the kwargs when creating the new version of the swath class.

Reply:
Sorry, I didn't understand the gist of your previous comments obviously. Removing.

Reply:
done, ready for rereview :)